app/models/compute_resources/foreman/model/ovirt.rb

```diff
@@ -295,6 +333,14 @@ def ca_cert
 
     private
 
+    def os_name_mapping(host)
+      host.operatingsystem.name =~ /redhat|centos/i ? 'rhel': host.operatingsystem.name
+    end
+
+    def arch_name_mapping(host)
+      host.architecture.name == 'x86_64' ? 'x64' : host.architecture.name if host.architecture
```

Comment:
Can we get this far and `host.architecture` be nil? I see `os_name_mapping` doesn't check for `.operatingsystem.present?` why should this one be different?